dotcom-rendering/src/server/dev-index.html

```diff
@@ -1037,69 +1052,135 @@ <h2>${title}</h2>
+				setHeadline(row, anchor) {
+					const headlineText = anchor.textContent;
+					const headline = row.querySelector('.headline');
+
+					if (headlineText !== null && headline !== null) {
+						headline.textContent = headlineText;
+					}
+				}
+
+				/**
+				 * @param {DocumentFragment} row
+				 * @param {HTMLAnchorElement} anchor
+				 * @return {void}
+				 */
+				setLinks(row, anchor) {
+					const href = anchor.getAttribute('href');
+
+					if (href !== null) {
+						row.querySelectorAll('article-link').forEach((link) => {
+							link.setAttribute('href', href);
+						});
+					}
+				}
+
+				/**
+				 * @param {HTMLLIElement} listItem
+				 * @return {DocumentFragment}
+				 */
+				tableRow(listItem) {
+					const row = this.rowTemplate.content.cloneNode(true);
+
+					if (!(row instanceof DocumentFragment)) {
+						throw new Error(
+							'article-examples: my row template content should be a DocumentFragment',
+						);
+					}
+
+					const anchor = listItem.querySelector('a');
+
+					if (!(anchor instanceof HTMLAnchorElement)) {
+						throw new Error(
+							'article-examples: each list item should have an anchor',
+						);
+					}
+
+					this.setHeadline(row, anchor);
+					this.setLinks(row, anchor);
+					this.setElementFromData(row, listItem, 'design');
+					this.setElementFromData(row, listItem, 'display');
+					this.setElementFromData(row, listItem, 'theme');
+
+					return row;
+				}
+
 				constructor() {
 					super();
 
 					const template = document.getElementById(
+						'articles-table-template',
+					);
+					const rowTemplate = document.getElementById(
 						'article-row-template',
 					);
 
 					if (!(template instanceof HTMLTemplateElement)) {
 						throw new Error(
-							"article-row: I couldn't find my template element",
+							"article-examples: I couldn't find my template element",
+						);
+					}
+
+					if (!(rowTemplate instanceof HTMLTemplateElement)) {
+						throw new Error(
+							"article-examples: I couldn't find my row template element",
 						);
 					}
 
 					const clone = template.content.cloneNode(true);
 
 					if (!(clone instanceof DocumentFragment)) {
 						throw new Error(
-							'article-row: my template content should be a DocumentFragment',
+							'article-examples: my template content should be a DocumentFragment',
 						);
 					}
 
 					this.content = clone;
+					this.rowTemplate = rowTemplate;
 				}
 
 				connectedCallback() {
-					const href = this.getAttribute('href');
-
-					if (href !== null) {
-						this.content
-							.querySelectorAll('article-link')
-							.forEach((link) => {
-								link.setAttribute('href', href);
-							});
-					}
-
-					this.setElementFromAttribute('headline');
-					this.setElementFromAttribute('design');
-					this.setElementFromAttribute('display');
-					this.setElementFromAttribute('theme');
+					const rows = Array.from(this.querySelectorAll('li')).map(
+						(li) => this.tableRow(li),
+					);
```

Comment:
```suggestion
					const rows = Array.from(this.querySelectorAll('li'))
						.map(this.tableRow);
```

Reply:
`tableRow` is a method and makes use of `this` internally. Unfortunately, passing it directly and allowing `map` to call it breaks the reference; `this` no longer refers to the class instance[^1][^2]:

```
Uncaught TypeError: this is undefined
```

It can be fixed in one of these two ways, but I think the existing call is easier to reason about?

```suggestion
					const rows = Array.from(this.querySelectorAll('li')).map(
						this.tableRow,
						this,
					);
```

```suggestion
					const rows = Array.from(this.querySelectorAll('li')).map(
						this.tableRow.bind(this),
					);
```

[^1]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/this#callbacks
[^2]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array#iterative_methods